### PR TITLE
Feature/eslint

### PR DESCRIPTION
**Pull request by dancrumb**
I'd like to be doing useful work on this and now that it's an NPM module, it may attract input from others too.

As such, I wanted to add ESLint support to the codebase. It's an effective linter and is highly configurable (so you don't have to waste time complying with rules that don't make sense).

This PR adds that and pulls a few things into shape. There's a little refactoring here and there, but all the tests still pass.

My next PR will be beefing up the tests, to make sure that I don't inadvertently break anything.

### CODE REVIEW

**Comment by dancrumb** [comment]
When reviewing, you can add `?w=1` to the URL to ignore whitespace only changes.

File: components/reverse.js [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
+    outport.send(this.createIP(reverseString(s)));
+    this.dropIP(ip);
+  }
+};
+
+// Thanks to Scott Gartner
+// Link to answer on StackOverflow: http://stackoverflow.com/a/17374133
+// Link to user profile page of Scott Gartner: http://stackoverflow.com/users/324657/scott-gartner
+// License: CC BY-SA 3.0 with attribution required: http://creativecommons.org/licenses/by-sa/3.0/
+function reverseString(str) {
```
Comment (dancrumb) [inline components/reverse.js]: JS convention is that functions in TitleCase should only be used for contsructors.

**Comment by jpaulm** [comment]
Hi Dan,

Have merged all PRs except #45.  Running `npm test` said `mocha`, `chai` and `mocha-fibers` were all missing, so I did `npm install`s for all of them, and `npm test` is now working - all passed with flying colours!  `node fbptest01` also works - didn't test the others...!  Shouldn't the three packages be hauled in automatically by something?!

PS Have reported the BitDefender problem to them! :-(

**Comment by dancrumb** [comment]
`npm test` doesn't run `npm install` by default, so if you don't run `npm install` explicitly first, you'll see it fail due to missing packages.

We could redefine `npm test` to run `npm install` by default - it's pretty much a no-op if someone's already done the package installs. However, it's a pretty overt failure mode with a straight forward resolution so... dealer's choice, I say.

**Pull request merged**